kaldb/src/main/java/com/slack/kaldb/chunk/manager/ChunkManager.java

```diff
@@ -0,0 +1,166 @@
+   * 1. topK results sorted by timestamp
+   * 2. histogram over a fixed time range
+   * We will not aggregate locally for future use-cases that have complex group by etc
+   */
+  public CompletableFuture<SearchResult<T>> query(SearchQuery query) {
+
+    SearchResult<T> errorResult =
+        new SearchResult<>(new ArrayList<>(), 0, 0, new ArrayList<>(), 0, 0, 1, 0);
+
+    List<CompletableFuture<SearchResult<T>>> queries =
+        chunkMap
+            .values()
+            .stream()
+            .filter(
+                chunk ->
+                    chunk.containsDataInTimeRange(
+                        query.startTimeEpochMs / 1000, query.endTimeEpochMs / 1000))
+            .map(
+                (chunk) ->
+                    CompletableFuture.supplyAsync(
+                            () -> {
+                              ScopedSpan span =
+                                  Tracing.currentTracer()
+                                      .startScopedSpan("ReadWriteChunkImpl.query");
+                              span.tag("chunkId", chunk.info().chunkId);
+                              span.tag(
+                                  "chunkCreationTimeSecsSinceEpoch",
+                                  String.valueOf(chunk.info().getChunkCreationTimeEpochSecs()));
+                              span.tag(
+                                  "chunkLastUpdatedTimeSecsEpochSecs",
+                                  String.valueOf(
+                                      chunk.info().getChunkLastUpdatedTimeSecsEpochSecs()));
+                              span.tag(
+                                  "dataStartTimeEpochSecs",
+                                  String.valueOf(chunk.info().getDataStartTimeEpochSecs()));
+                              span.tag(
+                                  "dataEndTimeEpochSecs",
+                                  String.valueOf(chunk.info().getDataEndTimeEpochSecs()));
+                              span.tag(
+                                  "chunkSnapshotTimeEpochSecs",
+                                  String.valueOf(chunk.info().getChunkSnapshotTimeEpochSecs()));
+                              span.tag("numDocs", String.valueOf(chunk.info().getNumDocs()));
+                              span.tag("chunkSize", String.valueOf(chunk.info().getChunkSize()));
+
+                              try {
+                                return chunk.query(query);
+                              } finally {
+                                span.finish();
+                              }
+                            },
+                            RequestContext.makeContextPropagating(queryExecutorService))
+                        // TODO: this will not cancel lucene query. Use ExitableDirectoryReader in
+                        // the future and pass this timeout
+                        .orTimeout(QUERY_TIMEOUT_SECONDS, TimeUnit.SECONDS))
```

Comment:
On this timeout would we call span.finish? Other wise the span will be flushed after 30 seconds and the span duration will be off.

Also, prefer the duration variant if available.

Reply:
This was reworked per feedback here - https://github.com/slackhq/kaldb/pull/122#discussion_r715819975. 

The `CompletableFuture` does not take a duration as a timeout so this approach is required.